doc/user/interop.md

```diff
@@ -89,137 +89,99 @@ Returns true only for the `nil` object.
 
 ### `HAS_KEYS`
 
-Returns true for all objects except those primitives that are always frozen,
-`nil`, `true`, `false`, `Fixnum`, `Bignum`, `Float`, `Symbol`. Note that `KEYS`
-will continue to return an empty list, even if `HAS_KEYS` returns false.
+Returns true.
 
 ### `KEYS`
 
-If the receiver is a Ruby `Hash`, return the hash keys.
+If the receiver is a Ruby `Array`, return an array of indices in
+bounds.
 
-`KEYS(hash)` → `hash.keys`
+Otherwise, if the receiver is a Ruby `Hash`, return the hash keys, via
+`receiver.keys`.
 
-Otherwise, return the instance variable names, without the leading `@`.
-
-`KEYS(other)` → `other.instance_variables.map { |key| ...key without @... }`
-
-In both cases if the `internal` flag is set, instance variable names with the
-leading `@` are also returned.
+Otherwise, return all methods via `receiver.methods`, and if the `internal` flag
```

Comment:
We could also consider returning only public methods, and including all methods only with internal flag. (KEY_INFO would have to update behavior as well)